curated_transformers/listener.py

```diff
@@ -144,6 +189,113 @@ def verify_inputs(self, inputs: Iterable[Doc]) -> bool:
                 return True
 
 
+class TransformerLayersListener(TransformerListener):
+    """Passes through the pooled representations of all transformer layers.
+
+    Requires its upstream Transformer component to return all layer outputs from
+    its model.
+    """
+
+    name = "transformer_layer_listener"
+
+    def __init__(
+        self,
+        upstream_name: str,
+        pooling: PoolingModelT,
+        width: int,
+        layers: int,
+        grad_factor: float,
+    ) -> None:
+        """
+        upstream_name (str):
+            A string to identify the 'upstream' Transformer component
+            to communicate with. The upstream name should either be the wildcard
+            string '*', or the name of the Transformer component. You'll almost
+            never have multiple upstream Transformer components, so the wildcard
+            string will almost always be fine.
+        pooling (PoolingModelT):
+            Model that is used to perform pooling over the piece representations.
+        width (int):
+            The width of the vectors produced by the upstream transformer component.
+        layers (int):
+            The the number of layers produced by the upstream transformer component,
+            excluding the embedding layer.
+        grad_factor (float):
+            Factor to multiply gradients with.
+        """
+        Model.__init__(
+            self,
+            name=self.name,
+            forward=tranformer_layers_listener_forward,
+            dims={"nO": width},
+            layers=[with_ragged_layers(pooling)],
+            attrs={
+                "grad_factor": grad_factor,
+                "layers": layers + 1,
+            },
+            refs={"pooling": pooling},
```

Comment:
Do we refer to this layer elsewhere in the codebase?

Reply:
Yeah, the distiller needs access to the pooler to apply it to the teacher outputs.